sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample04_PublishingEvents.md

```diff
@@ -48,13 +56,65 @@ finally
 
 ## Publishing events with automatic partition assignment
 
 Allowing automatic assignment to partitions is recommended when publishing needs to be highly available and shouldn't fail if a single partition is experiencing trouble.  Automatic assignment also helps to ensure that event data is evenly distributed among all available partitions, which helps to ensure throughput when publishing and reading data.
 
-When the batch is published, the `EventHubProducerClient` will receive an acknowledgment from the Event Hubs service; so long as no exception is thrown by this call, your application can consider publishing successful.  The service assumes responsibility for delivery of the batch.  All of your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
+### Event Hub Buffered Producer Client
+
+When using the `EventHubBufferedProducerClient`, events enqueued with no options specified will be automatically routed.  Because the producer manages publishing, there is no explicit call.  When the producer is closed, it will ensure that any remaining enqueued events have been published.  All of your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
+
+```C# Snippet:EventHubs_Sample04_AutomaticRoutingBuffered
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producer = new EventHubBufferedProducerClient(connectionString, eventHubName);
+
+// The failure handler is required and invoked after all allowable
+// retries were applied.
+
+producer.SendEventBatchFailedAsync += args =>
+{
+    Debug.WriteLine($"Publishing failed for { args.EventBatch.Count } events.  Error: '{ args.Exception.Message }'");
+    return Task.CompletedTask;
+};
+
+// The success handler is optional.
+
+producer.SendEventBatchSucceededAsync += args =>
+{
+   Debug.WriteLine($"{ args.EventBatch.Count } events were published to partition: '{ args.PartitionId }.");
+   return Task.CompletedTask;
+};
+
+try
+{
+    for (var index = 0; index < 5; ++index)
+    {
+        var eventData = new EventData($"Event #{ index }");
+        await producer.EnqueueEventAsync(eventData);
+    }
+}
+finally
+{
+    // Closing the producer will flush any
+    // enqueued events that have not been published.
+
+    await producer.CloseAsync();
+}
+```
+
+### Event Hub Producer Client
+
+When using the `EventHubProducerClient` a batch is first created and then published.  The `SendAsync` call will receive an acknowledgment from the Event Hubs service; so long as no exception is thrown, your application can consider publishing successful.  All of your event data will be published to one of the Event Hub partitions, thought here may be a slight delay until it is available to be read.
```

Comment:
```suggestion
When using the `EventHubProducerClient` a batch is first created and then published.  The `SendAsync` call will receive an acknowledgment from the Event Hubs service; so long as no exception is thrown, your application can consider publishing successful.  All of your event data will be published to one of the Event Hub partitions, though there may be a slight delay until it is available to be read.
```

Reply:
God... this one is just embarassing.

Reply:
😄 
![](https://media2.giphy.com/media/3orieKDJIFnmUIIY6Y/giphy.gif)